django/core/checks/translation.py

```diff
@@ -1,19 +1,68 @@
 from django.conf import settings
 from django.utils.translation.trans_real import language_code_re
 
-from . import Error, Tags, register
+from . import Error, Tags, Warning, register
 
 E001 = Error(
-    'You have provided an invalid value for the LANGUAGE_CODE setting.',
+    'You have provided an invalid value for the LANGUAGE_CODE setting: {!r}.',
```

Comment:
We're using `!r` for the invalid type cases yes?

Reply:
Good point :+1:  @pope1ni All tests pass with `{}` instead of '{!r}', so I think we can simplify messages and use `{}`.

Reply:
(But before we change it, what was the thinking?)

Reply:
We thought that for some unexpected non-string data types if may be better to return their representation. I think it is not necessary.

Reply:
Sorry. I think I was thinking of this so that invalid string values show up quoted. But not really all that important...